gateway/1.5/modules/ROOT/pages/policies-pdk-configure-features.adoc

```diff
@@ -1,5 +1,589 @@
+    let headers_state = response_state.into_headers_state().await;
+    
+    match config.dw_property.resolve_on_headers(&headers_state) {
+        Ok(value) => {
+            if value.is_null() {
+                logger::info!("Datewave expression resolved to null!");
+            }
+            else if let Some(val) = value.as_bool() {
+                logger::info!("Datewave expression resolved to boolean: {val}!");
+            }
+            else if let Some(val) = value.as_f64() {
+                logger::info!("Datewave expression resolved to f64: {val}!");
+            }
+            else if let Some(val) = value.as_str() {
+                logger::info!("Datewave expression resolved to str: {val}!");
+            }
+            else if let Some(val) = value.as_slice() {
+                logger::info!("Datewave expression resolved to slice: {val:?}!");
+            }
+            else if let Some(val) = value.as_object() {
+                logger::info!("Datewave expression resolved to object: {val:?}!");
+            }
+        },
+        Err(err) => {
+            logger::warn!("Failed to resolve the dataweave expression: {err}!")
+        }
+    }
+}
+----
+
+[[stop-request-execution]]
+== Stop Request Execution
+
+During request processing, you can abort the request and send a response.
+
+Do this by returning a `Response` object from your `on_request` wrapped functions:
+
+[source,Rust]
+----
+async fn request_filter(_request_state: RequestState) -> Response {
+    Response::new(401)
+        .with_headers(vec![("WWW-Authenticate".to_string(), "Bearer realm=\"oauth2\"".to_string())])
+        .with_body(r#"{ "error": "token was not present"}"#)
+}
+----
+
+This function is not very useful as it intercepts the requests no matter what. You can block or allow requests to reach the upstream service using a `Flow`. A `Flow` in an `enum` value with two possible values:
+
+* `Continue`: Lets the request pass. 
++
+This element defines an object that is forwarded to the response.
+* `Break`: Aborts the request and returns the provided response.
+
+Create a `Flow` as follows:
+
+[source,Rust]
+----
+async fn request_filter(request_state: RequestState) -> Flow<()> {
+    let header_state = request_state.into_headers_state().await;
+    if header_state.header("authorization").is_some() {
+        Flow::Continue(())
+    } else {
+        Flow::Break(Response::new(401)
+        .with_headers(vec![("WWW-Authenticate".to_string(), "Bearer realm=\"oauth2\"".to_string())])
+        .with_body(r#"{ "error": "token was not present"}"#))
+    }
+}
+----
+
+[IMPORTANT]
+====
+Due to the streaming nature of `proxy wasm`, it is possible for an early request to partially reach the upstream service while awaiting the `headers_state.into_body_state` function. The only way to avoid this by not awaiting the body. However, that is not possible for all policies. You should configure your upstream to ignore partial requests if the policy must read the body.
```

Comment:
```suggestion
Due to the streaming nature of `proxy wasm`, it is possible for an early request to partially reach the upstream service while awaiting the `headers_state.into_body_state` function. The only way to avoid this is by not awaiting the body. However, that is not possible for all policies. Configure your upstream to ignore partial requests if the policy must read the body.
```